### PR TITLE
Feat: Add path type filter

### PR DESCRIPTION
**Pull request by CaiJingLong**

### CODE REVIEW

File: example/lib/model/photo_provider.dart [hunk 1]
```diff
@@ -244,6 +245,38 @@ class PhotoProvider extends ChangeNotifier {
       thumbFormat = ThumbnailFormat.jpeg;
     }
   }
+
+  /// For path filter option
+  var _pathFilterOption = const PMPathFilter();
```
[the remark this answers is not in the document]
Comment (CaiJingLong) [reply example/lib/model/photo_provider.dart:250]: It is similar to 
```
// ==== next code is for path filter
```

It’s just for convenient positioning in the process of writing code, it doesn’t matter if you delete or keep it

File: lib/src/filter/path_filter.dart [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+import '../types/entity.dart';
+
+/// {@template PM.path_filter}
+///
+/// For filter the [AssetPathEntity].
+///
+/// Also see [PhotoManager.getAssetPathList]
+///
+/// {@endtemplate}
+class PMPathFilter {
```
Comment (AlexV525) [inline lib/src/filter/path_filter.dart:14]: These filters should be considered added to the README too.

**Comment by CaiJingLong** [comment]
Update README

The TOC of README was generated by idea.

**Review by AlexV525 — APPROVED** [review]

**Comment by emakar** [comment]
this fails web build with
```
/usr/lib/flutter/pub-cache/hosted/pub.dev/photo_manager-2.7.0/lib/src/filter/path_filter.dart:243:16:
Error: The integer literal 9223372036854775807 can't be represented exactly in JavaScript.
        return 9223372036854775807;
               ^^^^^^^^^^^^^^^^^^^
```

**Pull request merged**